x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_service.ts

```diff
@@ -223,6 +223,63 @@ export class EncryptedSavedObjectsService {
     return { attributes: clonedAttributes as T, error: decryptionError };
   }
 
+  public stripOrDecryptAttributesSync<T extends Record<string, unknown>>(
```

Comment:
__suggestion:__ my last ask is to refactor this a bit to share most of the logic with asynchronous version, something like this (feel free to change this snippet in any way you think is better). What do you think?
```ts
/**
 * Takes saved object attributes for the specified type and, depending on the type definition,
 * either decrypts or strips encrypted attributes (e.g. in case AAD or encryption key has changed
 * and decryption is no longer possible).
 * @param descriptor Saved object descriptor (ID, type and optional namespace)
 * @param attributesToStripOrDecrypt Object that includes a dictionary of __ALL__ saved object attributes stored
 * in Elasticsearch.
 * @param [originalAttributes] An optional dictionary of __ALL__ saved object original attributes
 * that were used to create that saved object (i.e. values are NOT encrypted).
 * @param [params] Parameters that control the way encrypted attributes are handled.
 */
public async stripOrDecryptAttributes<T extends Record<string, unknown>>(
  descriptor: SavedObjectDescriptor,
  attributesToStripOrDecrypt: T,
  originalAttributes?: T,
  params?: DecryptParameters
) {
  const { attributes, attributesToDecrypt } = this.prepareAttributesForStripOrDecrypt(
    descriptor,
    attributesToStripOrDecrypt,
    originalAttributes
  );
  try {
    const decryptedAttributes = attributesToDecrypt
      ? await this.decryptAttributes(descriptor, attributesToDecrypt, params)
      : {};
    return { attributes: { ...attributes, ...decryptedAttributes } };
  } catch (error) {
    return { attributes, error };
  }
}

/**
 * Takes saved object attributes for the specified type and, depending on the type definition,
 * either decrypts or strips encrypted attributes (e.g. in case AAD or encryption key has changed
 * and decryption is no longer possible).
 * @param descriptor Saved object descriptor (ID, type and optional namespace)
 * @param attributesToStripOrDecrypt Object that includes a dictionary of __ALL__ saved object attributes stored
 * in Elasticsearch.
 * @param [originalAttributes] An optional dictionary of __ALL__ saved object original attributes
 * that were used to create that saved object (i.e. values are NOT encrypted).
 * @param [params] Parameters that control the way encrypted attributes are handled.
 */
public stripOrDecryptAttributesSync<T extends Record<string, unknown>>(
  descriptor: SavedObjectDescriptor,
  attributesToStripOrDecrypt: T,
  originalAttributes?: T,
  params?: DecryptParameters
) {
  const { attributes, attributesToDecrypt } = this.prepareAttributesForStripOrDecrypt(
    descriptor,
    attributesToStripOrDecrypt,
    originalAttributes
  );
  try {
    const decryptedAttributes = attributesToDecrypt
      ? this.decryptAttributesSync(descriptor, attributesToDecrypt, params)
      : {};
    return { attributes: { ...attributes, ...decryptedAttributes } };
  } catch (error) {
    return { attributes, error };
  }
}

/**
 * Takes saved object attributes for the specified type and, depending on the type definition,
 * either strips encrypted attributes, replaces with original decrypted value if available, or
 * prepares them for decryption.
 * @private
 */
private prepareAttributesForStripOrDecrypt<T extends Record<string, unknown>>(
  descriptor: SavedObjectDescriptor,
  attributes: T,
  originalAttributes?: T
) {
  const typeDefinition = this.typeDefinitions.get(descriptor.type);
  if (typeDefinition === undefined) {
    return { attributes, attributesToDecrypt: null };
  }

  let attributesToDecrypt: T | undefined;
  const clonedAttributes: Record<string, unknown> = {};
  for (const [attributeName, attributeValue] of Object.entries(attributes)) {
    // We should strip encrypted attribute if definition explicitly mandates that.
    if (typeDefinition.shouldBeStripped(attributeName)) {
      continue;
    }

    // If attribute isn't supposed to be encrypted, just copy it to the resulting attribute set.
    if (!typeDefinition.shouldBeEncrypted(attributeName)) {
      clonedAttributes[attributeName] = attributeValue;
    } else if (originalAttributes) {
      // If attribute should be decrypted, but we have original attributes used to create object
      // we should get raw unencrypted value from there to avoid performance penalty.
      clonedAttributes[attributeName] = originalAttributes[attributeName];
    } else if (!attributesToDecrypt) {
      // Decrypt only attributes that are supposed to be exposed.
      attributesToDecrypt = Object.fromEntries(
        Object.entries(attributes).filter(([key]) => !typeDefinition.shouldBeStripped(key))
      ) as T;
    }
  }

  return {
    attributes: clonedAttributes as T,
    attributesToDecrypt:
      attributesToDecrypt && Object.keys(attributesToDecrypt).length > 0
        ? attributesToDecrypt
        : null,
  };
}
```

Reply:
Updated in https://github.com/elastic/kibana/pull/105968/commits/1365d05c60aeae8c4bd64962cfa6f93293dc4e0d